Building with `table-pager` fails after the new engine merge

**Describe the bug**
The table pager apparently uses async features which do not play with the new synchronous engine. Therefore, building nushell with `table-pager` feature fails.

**To Reproduce**
Steps to reproduce the behavior:
1. `cargo build --features=table-pager`
2. Observe errors related to futures

**Expected behavior**
Expected to build successfully 

**Configuration (please complete the following information):**

Run `version | pivot` and paste the output to show OS, features, etc.

```
> version | pivot
   0   version              0.29.2
   1   branch               main
   2   short_commit         42fac722
   3   commit_hash          42fac722bb67dc37f1db97f3406ac1a93a043c60
   4   commit_date          2021-04-06 23:14:06
   5   build_os             linux-x86_64
   6   rust_version         rustc 1.50.0 (cb75ad5db 2021-02-10)
   7   rust_channel         stable (default)
   8   cargo_version        cargo 1.50.0 (f04e7fab7 2021-02-04)
   9   pkg_version          0.29.2
  10   build_time           2021-04-07 10:56:36
  11   build_rust_channel   release
  12   features             clipboard-cli, ctrlc, default, directories, dirs, ichwh, ptree, rustyline, term,
                            trash, uuid, which, zip
```

I was just about to report this.
I used `git bisect` and can confirm the commit that causes it is 073e5727c6eab2adbe2c93cdd3042ac404218229

Full errors:
```
error[E0599]: no method named `add_exit_callback` found for opaque type `impl futures::Future` in the current scope
   --> crates/nu-command/src/commands/table/command.rs:207:26
    |
207 |             pager.lock().add_exit_callback(move || {
    |                          ^^^^^^^^^^^^^^^^^ method not found in `impl futures::Future`
    |
help: consider `await`ing on the `Future` and calling the method on its `Output`
    |
207 |             pager.lock().await.add_exit_callback(move || {
    |                          ^^^^^^

error[E0609]: no field `lines` on type `impl futures::Future`
   --> crates/nu-command/src/commands/table/command.rs:269:36
    |
269 |                     writeln!(pager.lines, "{}", output).map_err(|_| {
    |                                    ^^^^^ field not found in `impl futures::Future`

error[E0599]: no method named `data_finished` found for opaque type `impl futures::Future` in the current scope
   --> crates/nu-command/src/commands/table/command.rs:284:24
    |
284 |             pager_lock.data_finished();
    |                        ^^^^^^^^^^^^^ method not found in `impl futures::Future`
    |
help: consider `await`ing on the `Future` and calling the method on its `Output`
    |
284 |             pager_lock.await.data_finished();
    |                        ^^^^^^

error[E0277]: `std::result::Result<(), ShellError>` is not a future
   --> crates/nu-command/src/commands/table/command.rs:293:60
    |
293 |             join(minus::async_std_updating(pager.clone()), stream_data);
    |                                                            ^^^^^^^^^^^ `std::result::Result<(), ShellError>` is not a future
    | 
   ::: /home/s/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.13/src/future/join.rs:114:11
    |
114 |     Fut2: Future,
    |           ------ required by this bound in `futures::future::join`
    |
    = help: the trait `futures::Future` is not implemented for `std::result::Result<(), ShellError>`

error[E0277]: `std::result::Result<(), ShellError>` is not a future
   --> crates/nu-command/src/commands/table/command.rs:293:13
    |
293 |               join(minus::async_std_updating(pager.clone()), stream_data);
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::result::Result<(), ShellError>` is not a future
    | 
   ::: /home/s/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.13/src/future/join.rs:76:1
    |
76  | / generate! {
77  | |     /// Future for the [`join`](join()) function.
78  | |     (Join, <Fut1, Fut2>),
79  | |
...   |
87  | |     (Join5, <Fut1, Fut2, Fut3, Fut4, Fut5>),
88  | | }
    | |_- required by this bound in `futures::future::Join`
    |
    = help: the trait `futures::Future` is not implemented for `std::result::Result<(), ShellError>`

error[E0308]: mismatched types
   --> crates/nu-command/src/commands/table/command.rs:292:13
    |
292 |         let (minus_result, streaming_result) =
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `futures::future::Join`, found tuple
293 |             join(minus::async_std_updating(pager.clone()), stream_data);
    |             ----------------------------------------------------------- this expression has type `futures::future::Join<impl futures::Future, std::result::Result<(), ShellError>>`
    | 
   ::: /home/s/.cargo/registry/src/github.com-1ecc6299db9ec823/minus-3.3.0/src/rt_wrappers/async_std_wrapper.rs:54:60
    |
54  | pub async fn async_std_updating(pager: Arc<PagerMutex>) -> Result<(), AlternateScreenPagingError> {
    |                                                            -------------------------------------- the `Output` of this `async fn`'s expected opaque type
    |
    = note: expected struct `futures::future::Join<impl futures::Future, std::result::Result<(), ShellError>>`
                found tuple `(_, _)`

error[E0277]: `std::result::Result<(), ShellError>` is not a future
   --> crates/nu-command/src/commands/table/command.rs:293:13
    |
168 |       let mut args = args.evaluate_once()?;
    |                      --------------------- this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
...
179 |       let mut start_number = if let Some(f) = args.get_flag("start_number")? {
    |                                               ------------------------------ this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
...
269 | /                     writeln!(pager.lines, "{}", output).map_err(|_| {
270 | |                         ShellError::untagged_runtime_error("Error writing to pager")
271 | |                     })?;
    | |_______________________- this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
...
293 |               join(minus::async_std_updating(pager.clone()), stream_data);
    |               ^^^^ `std::result::Result<(), ShellError>` is not a future
294 |           minus_result.map_err(|_| ShellError::untagged_runtime_error("Error paging data"))?;
    |           ---------------------------------------------------------------------------------- this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
295 |           streaming_result?;
    |           ----------------- this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
...
301 |       Ok(OutputStream::empty())
    |       ------------------------- this returned value is of type `std::result::Result<nu_stream::InputStream, ShellError>`
    |
    = help: within `futures::future::Join<impl futures::Future, std::result::Result<(), ShellError>>`, the trait `futures::Future` is not implemented for `std::result::Result<(), ShellError>`
    = note: required because it appears within the type `futures::future::Join<impl futures::Future, std::result::Result<(), ShellError>>`
    = note: the return type of a function must have a statically known size

warning: unused import: `std::fmt::Write`
  --> crates/nu-command/src/commands/table/command.rs:17:5
   |
17 |     std::fmt::Write,
   |     ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error: aborting due to 7 previous errors; 1 warning emitted

Some errors have detailed explanations: E0277, E0308, E0599, E0609.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `nu-command`

```

--- a/crates/nu-command/src/commands/table/command.rs
+++ b/crates/nu-command/src/commands/table/command.rs
@@ -1,6 +1,7 @@
 use crate::commands::table::options::{ConfigExtensions, NuConfig};
 use crate::prelude::*;
 use crate::primitive::get_color_config;
+use futures::executor::block_on;
 use nu_data::value::{format_leaf, style_leaf};
 use nu_engine::WholeStreamCommand;
 use nu_errors::ShellError;
@@ -194,7 +195,7 @@ fn table(args: CommandArgs) -> Result<OutputStream, ShellError> {
         .set_input_handler(Box::new(input_handling::MinusInputHandler {}))
         .finish();
 
-    let stream_data = {
+    let stream_data = async {
         let finished = Arc::new(AtomicBool::new(false));
         // we are required to clone finished, for use within the callback, otherwise we get borrow errors
         #[cfg(feature = "table-pager")]
@@ -204,7 +205,7 @@ fn table(args: CommandArgs) -> Result<OutputStream, ShellError> {
             // This is called when the pager finishes, to indicate to the
             // while loop below to finish, in case of long running InputStream consumer
             // that doesn't finish by the time the user quits out of the pager
-            pager.lock().add_exit_callback(move || {
+            pager.lock().await.add_exit_callback(move || {
                 finished_within_callback.store(true, Ordering::Relaxed);
             });
         }
@@ -265,7 +266,7 @@ fn table(args: CommandArgs) -> Result<OutputStream, ShellError> {
                 let output = draw_table(&t, term_width, &color_hm);
                 #[cfg(feature = "table-pager")]
                 {
-                    let mut pager = pager.lock();
+                    let mut pager = pager.lock().await;
                     writeln!(pager.lines, "{}", output).map_err(|_| {
                         ShellError::untagged_runtime_error("Error writing to pager")
                     })?;
@@ -280,7 +281,7 @@ fn table(args: CommandArgs) -> Result<OutputStream, ShellError> {
 
         #[cfg(feature = "table-pager")]
         {
-            let mut pager_lock = pager.lock();
+            let mut pager_lock = pager.lock().await;
             pager_lock.data_finished();
         }
 
@@ -290,13 +291,14 @@ fn table(args: CommandArgs) -> Result<OutputStream, ShellError> {
     #[cfg(feature = "table-pager")]
     {
         let (minus_result, streaming_result) =
-            join(minus::async_std_updating(pager.clone()), stream_data);
+            block_on(join(minus::async_std_updating(pager.clone()), stream_data));
         minus_result.map_err(|_| ShellError::untagged_runtime_error("Error paging data"))?;
         streaming_result?;
     }
 
     #[cfg(not(feature = "table-pager"))]
-    stream_data.map_err(|_| ShellError::untagged_runtime_error("Error streaming data"))?;
+    block_on(stream_data)
+        .map_err(|_| ShellError::untagged_runtime_error("Error streaming data"))?;
 
     Ok(OutputStream::empty())
 }

